docs/matchers/watson.md

```diff
@@ -0,0 +1,161 @@
+# IBM Watson Matcher
+
+[IBM Watson](https://www.ibm.com/watson) is an NLP API for matching strings to [intents](https://cloud.ibm.com/docs/services/assistant?topic=assistant-intents). Intents are created on your assistant intent page.
+
+## Configuring opsdroid
+
+In order to enable Watson skills, you must specify an `access-token`, an `assistant-id` and a `gateway` for your bot in the parsers section of the opsdroid configuration file.
+You can find these informations inside your Assistant Settings under the `API Details`. Note that depending where your bot is located the gateway will be different, just use this. For example: `gateway-fra`.
+
+You can also set a `min-score` option to tell opsdroid to ignore any matches which score less than a given number between 0 and 1. The default for this is 0 which will match all messages.
+
+```yaml
+
+parsers:
+  - name: watson
+    gateway: 'gateway-fr' # Required
+    access-token: XJF475SKGITJ98KHFO # Required
+    assistant-id: '74yhfhis9-kfirj1e-jfir34-kfdir345' # Required
+    min-score: 0.6
+```
+
+### Localization
+
+If you want to use Watson on a different language you will have to create different intents and entities to handle the languages that you wish to support.
+
+## Using the parser with a skill
+
+In this example we have a Watson assistant set up with the default Customer Care Sample Skill - this will show you how you can get entities and replies from the assistant and include them with opsdroid.
+
+### [Example 1](#example1)
+
+```python
+from opsdroid.skill import Skill
+from opsdroid.matchers import match_watson
+import logging
+
+_LOGGER = logging.getLogger(__name__)
+
+class MySkill(Skill):
+    @match_watson('Customer_Care_Appointments')
+    async def book_slot(self, message):
+        """Book an appointment"""
+        reply = message.watson['output']['generic'][0]['text']
```

Comment:
What is this `reply` object? I see you log it below but don't do anything else with it.

Reply:
I was planning to do something with the reply but then didn't, will add another example that uses the reply it will make more sense I think

Reply:
Ok 👍